community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/ShortestPathComplexQueryAcceptanceTest.scala

```diff
@@ -32,42 +32,33 @@ class ShortestPathComplexQueryAcceptanceTest extends ExecutionEngineFunSuite wit
     setupModel()
     val result = executeUsingCostPlannerOnly(
       """
-        |//same idea but iterating over two collections of nodes
-        |profile match (charles:Pixie { fname : 'Charles'}),(joey:Pixie { fname : 'Joey'}),(kim:Pixie { fname : 'Kim'})
-        |with kim as kimDeal, collect(charles) as charlesT, collect(joey) as joeyS
-        |unwind charlesT AS charlesThompson
-        |unwind joeyS AS joeySantiago
-        |match pathx = allShortestPaths((charlesThompson)-[*1..5]-(joeySantiago))
-        |where none (n IN nodes(pathx) where id(n) = id(kimDeal))
-        |with nodes(pathx) as nodes
-        |return nodes
+        |PROFILE MATCH (charles:Pixie { fname : 'Charles'}),(joey:Pixie { fname : 'Joey'}),(kim:Pixie { fname : 'Kim'})
+        |WITH kim AS kimDeal, collect(charles) AS charlesT, collect(joey) AS joeyS
+        |UNWIND charlesT AS charlesThompson
+        |UNWIND joeyS AS joeySantiago
+        |MATCH pathx = allShortestPaths((charlesThompson)-[*1..5]-(joeySantiago))
+        |WHERE none (n IN nodes(pathx) WHERE id(n) = id(kimDeal))
+        |RETURN extract(node in nodes(pathx) | id(node)) as ids
```

Comment:
Not saying you need to change this, but the fancy way of doing extract is through list comprehensions: `[node IN nodes(pathx) | id(node)] AS ids`